services/criterion/criterion.service.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { BaseJsonService } = require('..')
+
+const IMPROVED = 'improved'
+
+function errorMessagesFor(notFoundMessage = 'status not found') {
+  return {
+    404: notFoundMessage,
+    422: notFoundMessage,
+  }
+}
+
+const schema = Joi.string().required()
+
+module.exports = class Criterion extends BaseJsonService {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get route() {
+    return {
+      base: 'criterion',
+      pattern: ':user/:repo',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Criterion',
+        namedParams: {
+          user: 'chmoder',
+          repo: 'data_vault',
+        },
+        staticPreview: this.render({ isImproved: true, status: IMPROVED }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'criterion' }
+  }
+
+  static render({ isImproved, status }) {
+    if (isImproved) {
+      return {
+        message: `${status}`,
+        color: 'brightgreen',
+      }
+    } else {
+      return {
+        message: `${status}`,
+        color: 'yellow',
+      }
+    }
+  }
+
+  async handle({ user, repo }) {
+    const status = await this._requestJson({
```

Comment:
> Badges should not obtain data from undocumented or reverse-engineered API endpoints.

I see, this is a little unfortunate but also understandable.  I am basically just promoting this idea from from their docs: [Performance has improved.
](https://bheisler.github.io/criterion.rs/book/user_guide/command_line_output.html)

I definitely need and want documentation, so I can make that a priority.  I don't think we should make an exception to the guidelines for my particular badge.

Reply:
One of the main drivers for this is the myriad upstream services/tools. that Shields integrates with to provide badges for and the small group of Shields maintainers. We're definitely not experts nor intimately familiar with every service and their respective API contracts, so it's really important that we have things like reference material, docs, contact methods, etc. 

Even something as simple as a Swagger endpoint and an open repo for issues/contact on GitHub/Gitlab/Bitbucket/wherever would be a great starting point.

Reply:
I actually made this [swagger endpoint](https://app.swaggerhub.com/apis-docs/chmoder/Criterion.dev/1.0.0) last night last night.  I was holding off to show you till I added the complete set of available routes; but at this time it has the status route documented.

Right now the repo is private, GitHub has a solution i'm not a fan of, to make an [issues only repo](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-an-issues-only-repository).  I just need to host my authentication information (database connections) somewhere else before I can make it public.

It has been great collaborating with you both, and a good learning experience.  I will follow up on this issue when these things are ready for review.

Reply:
Hello @calebcartwright!  

I decided to open an issues only repo after all because I actually do want to keep the service private.  I'm still glad I moved the sensitive data out of the repository, so that wasn't lost time.  Below are the requested resources, let me know if there is anything else I can do.

Support and contact:
https://github.com/chmoder/api.criterion.dev

API documentation:
https://app.swaggerhub.com/apis-docs/chmoder/Criterion.dev

Reply:
Looks good to me, let's see what @calebcartwright thinks.

Maybe worth putting the GitHub repo link (which itself links to the API docs) in a comment in the badge code directly?